### PR TITLE
feat(paper checkout): Add support for Paper Checkout 

### PR DESCRIPTION
**Pull request by ElasticBottle**

### CODE REVIEW

File: src/interfaces/PaperSDKError.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ export type PaperSDKError = {
 };
 
 export enum PaperSDKErrorCode {
+  UserCancelledOperation = 'User Cancelled Operation',
```
Comment (arcoraven) [inline src/interfaces/PaperSDKError.ts:11]: Nit: This is pretty unclear and vague. Let's just make it explicit?

`UserAbandonedCheckout = 'User abandoned the checkout.'`

File: src/lib/CheckoutComponent.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
+import { PAPER_APP_URL } from '../constants/settings';
```
Comment (arcoraven) [inline src/lib/CheckoutComponent.ts:1]: Can we call this file CheckoutLink? CheckoutComponent makes it sound like it's for CheckoutElement

File: src/lib/CheckoutComponent.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            break;
+          }
+          case 'claimSuccessful': {
+            const { id: transactionId, claimedTokens } = e.data;
+            PaperCheckoutResult.transactionId = transactionId;
+            PaperCheckoutResult.claimedTokens = claimedTokens;
+            // let the user see the completed NFT for some short time before closing it
+            await sleepForSeconds(3.5);
+            resolve(result);
+            drawer.close();
```
Comment (arcoraven) [inline src/lib/CheckoutComponent.ts:64]: Would this have been simpler:
```typescript
setTimeout(() => {
  resolve(result);
  drawer.close();
}, 3500);
```

File: src/interfaces/CheckoutSuccessType.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export type CheckoutSuccessType = {
```
Comment (arcoraven) [inline src/interfaces/CheckoutSuccessType.ts:1]: The naming is a bit vague and "Type" shouldn't be in there. Maybe `CheckoutSuccessResult`?

File: src/lib/CheckoutComponent.ts (hunk 2 above)
Comment (ElasticBottle) [reply src/lib/CheckoutComponent.ts:1]: Do you think just naming it `renderPaperCheckoutLink` is a bad idea? `CheckoutLink` feels like it's suggesting something else as well 🤔

File: src/lib/CheckoutComponent.ts (hunk 3 above)
Comment (ElasticBottle) [reply src/lib/CheckoutComponent.ts:64]: I find `sleepForSeconds` more declarative and clearer.

**Pull request merged**